esmvaltool/recipes/recipe_ocean_Landschutzer2016.yml

```diff
@@ -0,0 +1,124 @@
+# ESMValTool
+# Landschutzer2014.yml
+---
+documentation:
+  description: |
+    Recipe to evaluate CO2 Fluxes of marine biogeochemistry models of CMIP5
+    using Landschutzer et al (2.014) dataset.
+    Written by Tomas Lovato, CMCC, tomas.lovato@cmcc.it
+
+  authors:
+    - lova_to
+  
+  maintainer:
+    - lova_to
+
+  references:
+    - acknow_project
+
+  projects:
+    - crescendo
+
+
+datasets:
+# working datasets
+  - {dataset: CanESM2,       project: CMIP5,  exp: historical,  ensemble: r1i1p1,  start_year: 2001,  end_year: 2004}
+
+
+# --------------------------------------------------
+# Preprocessors
+# --------------------------------------------------
+preprocessors:
+  # --------------------------------------------------
+  # map preprocessors
+  # --------------------------------------------------
+  # For a 2D global surface map
+  prep_surface_map_2D:
+    time_average:
+    regrid:
+      target_grid: 2x2
+      scheme: linear
+
+  # Global area-weighted Average from 2D field
+  prep_global_Surface_average_timeseries_2D:
+    custom_order: true
+    average_region:
+      coord1: longitude
+      coord2: latitude
+    multi_model_statistics:
+      span: overlap
+      statistics: [mean ]
+
+# --------------------------------------------------
+# Diagnostics
+# --------------------------------------------------
+diagnostics:
+
+  # --------------------------------------------------
+  # Surface time series vs OBS
+  # --------------------------------------------------
+  diag_timeseries_surface_average_vs_OBS:
+    description: Global surface time series (Landschutzer2014 observations)
+    variables:
+      dpco2:
+        preprocessor: prep_global_Surface_average_timeseries_2D
+        field: TO2M
```

Comment:
Field is no longer required, #775, can be removed

Reply:
@bouweandela I merged my local code with the last version2_development and removed the field from the recipe.
It does not work, code asks for the missing `field` required by `recipe_schema.yml`

Should I just remove field from this recipe and then other changes will occur later on?

Reply:
You can remove field also from the schema

Reply:
@bouweandela I correct the schema and now I got the following 

`esmvaltool._recipe_checks.RecipeError: Missing keys {'field'} from variable dpco2 in diagnostic diag_timeseries_surface_average_vs_OBS`

The solution is to remove it from here 
https://github.com/ESMValGroup/ESMValTool/blob/4e017b941110a6375c9cc91cb8040c3e5569e054/esmvaltool/_recipe.py#L875-L878

Note that in `_recipe.py` there are several hardcoded references to `field`

Once this is corrected You get an error in  `_data_finder.py` 
```
  File "/users/home/ans033/GIT/ESMValTool/esmvaltool/_recipe.py", line 767, in _get_preprocessor_task
    name=task_name)
  File "/users/home/ans033/GIT/ESMValTool/esmvaltool/_recipe.py", line 636, in _get_single_preprocessor_task
    config_user=config_user,
  File "/users/home/ans033/GIT/ESMValTool/esmvaltool/_recipe.py", line 579, in _get_preprocessor_products
    config_user['preproc_dir'])
  File "/users/home/ans033/GIT/ESMValTool/esmvaltool/_data_finder.py", line 280, in get_output_file
    _replace_tags(cfg['output_file'], variable)[0] + '.nc',
  File "/users/home/ans033/GIT/ESMValTool/esmvaltool/_data_finder.py", line 117, in _replace_tags
    "your recipe entry".format(tag, variable))
KeyError: "Dataset key field must be specified for {'preprocessor': 'prep_global_Surface_average_timeseries_2D', 'mip': 'Omon', 'fx_files': {'areacello': '/work/ans033/esmvaltool/CMIP5/CanESM2/fx/areacello_fx_CanESM2_historical_r0i0p0.nc'}, 'variable_group': 'dpco2', 'short_name': 'dpco2', 'diagnostic': 'diag_timeseries_surface_average_vs_OBS', 'dataset': 'CanESM2', 'project': 'CMIP5', 'exp': 'historical', 'ensemble': 'r1i1p1', 'start_year': 2001, 'end_year': 2004, 'recipe_dataset_index': 0, 'cmor_table': 'CMIP5', 'institute': ['CCCma'], 'standard_name': 'surface_carbon_dioxide_partial_pressure_difference_between_sea_water_and_air', 'long_name': 'Delta PCO2', 'units': 'Pa', 'modeling_realm': ['ocnBgchem'], 'frequency': 'mon'}, check your recipe entry"
```
It is necessary to modify _data_finder.py
https://github.com/ESMValGroup/ESMValTool/blob/4e017b941110a6375c9cc91cb8040c3e5569e054/esmvaltool/_data_finder.py#L286-L293

and also remove field from `config-developer.yml`
https://github.com/ESMValGroup/ESMValTool/blob/4e017b941110a6375c9cc91cb8040c3e5569e054/esmvaltool/config-developer.yml#L167-L169

now the recipe is working without `field`, but I suspect that more testing would be necessary before such changes.

If you are ok I can commit all the above.

Reply:
My apologies for getting you into this mess, I thought it would be easier to remove. I'll create a separate pull request for this.

Reply:
@bouweandela So far the recipe work for CMIP5 data, but it requires testing with CMIP6 ones. I committed the required changes to this branch (it is 4 rows of code in total), probably it will be better to integrate those changes and start from here. I really would like to close the PR that I opened 2 weeks ago ...